File: update_footer.py

import os
import re

# The complete footer and modals block from index.html (lines 627-843 approx)
# I will read it from the file to be sure I have the latest version.

with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()
    # Find the footer start
    footer_match = re.search(r'<footer class="py-16 md:py-24 bg-surface border-t border-white/5">', content)
    if footer_match:
        # Get everything from footer start to </body>
        footer_block = content[footer_match.start():content.rfind('</body>')]
    else:
        print("Footer not found in index.html")
        exit(1)

html_files = [f for f in os.listdir('.') if f.endswith('.html')]

for file_name in html_files:
    if file_name == 'index.html':
        continue
        
    print(f"Processing {file_name}...")
    with open(file_name, 'r', encoding='utf-8') as f:
        content = f.read()

    # Step 1: Remove the specific text from plataforma.html (or any other)
    if file_name == 'plataforma.html':
        content = re.sub(r'<p class="mt-8 text-xs text-slate-600 uppercase tracking-widest font-bold font-editorial italic">\s*\* Lanzamiento total Q3 2026\. Acceso beta disponible para inversores\.</p>', '', content)
        # Also handle potential variations in spacing
        content = re.sub(r'\* Lanzamiento total Q3 2026\. Acceso beta disponible para inversores\.', '', content)

    # Step 2: Replace footer
    # Find the existing footer
    existing_footer_match = re.search(r'<footer.*?>.*?</footer>', content, re.DOTALL)
    if existing_footer_match:
        # Find all trailing content after the footer (like modals or just whitespace)
        # and replace it with the new footer block
        new_content = content[:existing_footer_match.start()] + footer_block + "\n</body>\n</html>"
        
        # Check if there was multiple footers or weird nesting
        # Actually, let's just replace from the first <footer to the end of the file (before </html>)
        new_content = re.sub(r'<footer.*', footer_block + "\n</body>\n</html>", content, flags=re.DOTALL)
        
        with open(file_name, 'w', encoding='utf-8') as f:
            f.write(new_content)
    else:
        print(f"No footer found in {file_name}, skipping replacement but check manually if needed.")

print("Done!")

File: update_footer.js

const fs = require('fs');
const path = require('path');

const indexContent = fs.readFileSync('index.html', 'utf8');
const footerTag = '<footer class="py-16 md:py-24 bg-surface border-t border-white/5">';
const footerStart = indexContent.indexOf(footerTag);
const bodyEnd = indexContent.lastIndexOf('</body>');

if (footerStart === -1 || bodyEnd === -1) {
    console.error('Footer not found in index.html');
    process.exit(1);
}

const footerBlock = indexContent.substring(footerStart, bodyEnd).trim();
const htmlFiles = fs.readdirSync('.').filter(f => f.endsWith('.html') && f !== 'index.html');

htmlFiles.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');

    // Remove the specific text
    // Searching for various versions of it
    const textToRemove = [
        /\* Lanzamiento total Q3 2026\. Acceso beta disponible para inversores\./g,
        /<p class="mt-8 text-xs text-slate-600 uppercase tracking-widest font-bold font-editorial italic">\s*\* Lanzamiento total Q3 2026\. Acceso beta disponible para inversores\.\s*<\/p>/g
    ];

    textToRemove.forEach(re => {
        content = content.replace(re, '');
    });

    // Replace everything from the first <footer to the end of the file
    // We want to keep everything up to the first footer, then put the new footer block, then body and html closing tags
    const firstFooterIndex = content.indexOf('<footer');
    if (firstFooterIndex !== -1) {
        const headPart = content.substring(0, firstFooterIndex);
        const newContent = headPart + "\n    " + footerBlock + "\n\n</body>\n</html>";
        fs.writeFileSync(file, newContent);
        console.log(`Updated ${file}`);
    } else {
        console.warn(`No footer found in ${file}`);
    }
});
